app/src/androidTest/java/com/github/sdpcoachme/MainActivityTest.kt

```diff
@@ -1,34 +1,136 @@
 package com.github.sdpcoachme
 
+import android.content.Intent
 import androidx.compose.ui.test.*
-import androidx.compose.ui.test.junit4.createAndroidComposeRule
-import androidx.test.espresso.Espresso
-import androidx.test.espresso.intent.Intents
-import androidx.test.espresso.intent.Intents.intended
-import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
-import androidx.test.espresso.intent.matcher.IntentMatchers.hasExtra
+import androidx.test.espresso.matcher.ViewMatchers
 import androidx.test.ext.junit.runners.AndroidJUnit4
-import org.hamcrest.Matchers.allOf
-import org.junit.Rule
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.*
+import com.google.firebase.auth.FirebaseAuth
+import junit.framework.TestCase
+import org.hamcrest.CoreMatchers
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
-class MainActivityTest {
+open class MainActivityTest {
 
-    @get:Rule
-    val composeTestRule = createAndroidComposeRule<MainActivity>()
+//    @get:Rule
+//    val composeTestRule = createAndroidComposeRule<MainActivity>()
+//
+//    @Test
+//    fun correctIntentFiredOnDisplayMessageClick() {
+//        Intents.init()
+//
+//        val name = "John Lennon"
+//        composeTestRule.onNodeWithTag("textfield").performTextInput(name)
+//        Espresso.closeSoftKeyboard()
+//        composeTestRule.onNodeWithTag("button").performClick()
+//
+//        intended(allOf(hasComponent(GreetingActivity::class.java.name), hasExtra("name", name)))
+//        Intents.release()
+//    }
```

Comment:
I think we can delete the commented correctIntentFiredOnDisplayMessageClick() test function !